### PR TITLE
Add scale-ci-watcher job

### PR DESCRIPTION
**Pull request by chaitanyaenr**
This commit:
- Adds a template, build script and properties file for scale-ci-watcher
  to create/update the jobs. This job is enabled by default and also can
  be kicked off at anytime to update the jobs when a change gets commited
  into the scale-ci-pipeline repo.
- updates readme with the status.

### CODE REVIEW

File: jjb/dynamic/scale-ci_watcher.yml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    - shell: |-
+        set -exo pipefail
+
+        # Delete if the repo already exists
+        if [[ -d "$SCALE_CI_WATCHER_REPO_PATH" ]]; then
+          rm -rf  $SCALE_CI_WATCHER_REPO_PATH
+        fi
+
+        # Fetch the code
+        git clone $SCALE_CI_WATCHER_REPO $SCALE_CI_WATCHER_REPO_PATH
```
Comment (akrzos) [inline jjb/dynamic/scale-ci_watcher.yml]: Can this be managed by the scm module?  The module allows parameters I believe.

Comment (chaitanyaenr) [reply jjb/dynamic/scale-ci_watcher.yml]: Yes, I will modify it to be managed by the git plugin.

**Comment by chaitanyaenr** [comment]
@akrzos the scale-ci-pipeline repo clone is managed by git plugin now. PTAL. Thanks.

File: jjb/dynamic/scale-ci_watcher.yml [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        clean-parent: false
+        dirmatch: false
+        exclude: []
+        fail-build: true
+        include: []
+    scm:
+    - git:
+        branches:
+        - '*/master'
+        url: https://github.com/openshift-scale/scale-ci-pipeline.git
```
Comment (akrzos) [inline jjb/dynamic/scale-ci_watcher.yml]: Should this be $SCALE_CI_WATCHER_REPO?

File: jjb/dynamic/scale-ci_watcher.yml [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        - aborted: true
+        clean-parent: false
+        dirmatch: false
+        exclude: []
+        fail-build: true
+        include: []
+    scm:
+    - git:
+        branches:
+        - '*/master'
```
Comment (akrzos) [inline jjb/dynamic/scale-ci_watcher.yml]: Should this be $SCALE_CI_WATCHER_REPO_BRANCH?

Comment (chaitanyaenr) [reply jjb/dynamic/scale-ci_watcher.yml]: @akrzos fixed it, thanks. PTAL.

**Pull request merged**